searx/preferences.py

```diff
@@ -185,34 +177,34 @@ def parse(self, data):
 class MapSetting(Setting):
     """Setting of a value that has to be translated in order to be storable"""
 
-    def _post_init(self):
-        if not hasattr(self, 'map'):
-            raise MissingArgumentException('missing argument: map')
-        if self.value not in self.map.values():  # pylint: disable=no-member
+    def __init__(self, default_value, map: Dict[str, object], locked=False):  # pylint: disable=redefined-builtin
```

Comment:
Nit: map is a keyword .. I often at the same situation where a name that fits is an keyword e.g. `map` or `dict` .. I usually then choose a name like `kv` (key/value) ... `kvmap` or similar .. from my side `map` is OK here / no need to change, I just wanted to leave a note about my thoughts.

Reply:
nit: `map` isn't a keyword ... it's just a built-in function (otherwise this would be a syntax error).

Yeah I'm currently not really interested in fixing such pylint errors.

Reply:
> nit: map isn't a keyword ... it's just a built-in function (otherwise this would be a syntax error).

right. Sorry I'm fighting with my github editor .. I edited my initial comment ..

Reply:
>map is a keyword

[No it's not.](https://docs.python.org/3/reference/lexical_analysis.html#keywords) as I explained it's a [built-in function](https://docs.python.org/3/library/functions.html).

The difference is that `map = 1` works while `for = 1` is a syntax error.